operator/controllers/common/phasehandler.go

```diff
@@ -106,3 +111,81 @@ func (r PhaseHandler) HandlePhase(ctx context.Context, ctxTrace context.Context,
 
 	return &PhaseResult{Continue: false, Result: requeueResult}, nil
 }
+
+func (r PhaseHandler) createFailureReasonSpanEvents(ctx context.Context, phase common.KeptnPhaseType, object *PhaseItemWrapper, spanTrace trace.Span) error {
+	var messages []string
+	var err error
+	if phase.IsEvaluation() {
+		messages, err = r.GetEvaluationFailureReasons(ctx, phase, object)
+	} else if phase.IsTask() {
+		messages, err = r.GetTaskFailureReasons(ctx, phase, object)
+	}
+
+	if err != nil {
+		return err
+	}
+
+	for _, msg := range messages {
+		spanTrace.AddEvent(msg)
```

Comment:
for tasks we would know when this would happen https://github.com/odubajDT/lifecycle-controller/blob/feat/396/fail-detail-span/operator/api/v1alpha1/keptntask_types.go#L69
Maybe we could add the `task.EndTime()` as timestamp of the event